qa/util/src/main/java/io/camunda/qa/util/multidb/CamundaMultiDBExtension.java

```diff
@@ -301,34 +310,121 @@ public void beforeAll(final ExtensionContext context) {
+                  + "/realms/camunda/protocol/openid-connect/token");
+      injectStaticKeycloakContainerField(testClass, keycloakContainer);
+    } else {
+      authenticatedClientFactory = new BasicAuthCamundaClientTestFactory(applicationUnderTest);
+    }
+
     if (applicationUnderTest.shouldBeManaged) {
       manageApplicationUnderTest();
     }
 
-    authenticatedClientFactory = new CamundaClientTestFactory(applicationUnderTest);
-    entityManager = new EntityManager(authenticatedClientFactory.getDefaultUserCamundaClient());
-    createEntities(testClass);
+    entityManager = new EntityManager(authenticatedClientFactory.getAdminCamundaClient());
+    createEntities(testClass, shouldSetupKeycloak);
+
     // we support only static fields for now - to make sure test setups are build in a way
     // such they are reusable and tests methods are not relying on order, etc.
     // We want to run tests in an efficient manner, and reduce setup time
     injectStaticClientField(testClass);
   }
 
-  private void createEntities(final Class<?> testClass) {
+  private KeycloakContainer setupKeycloak() {
+    keycloakContainer = DefaultTestContainers.createDefaultKeycloak();
+    closeables.add(keycloakContainer);
+    keycloakContainer.start();
+
+    final var realm = new RealmRepresentation();
+    realm.setRealm(KEYCLOAK_REALM);
+    realm.setEnabled(true);
+    try (final var keycloak = keycloakContainer.getKeycloakAdminClient()) {
+      keycloak.realms().create(realm);
+    }
+    setupUserInKeycloak(
+        TestStandaloneBroker.DEFAULT_MAPPING_ID, TestStandaloneBroker.DEFAULT_MAPPING_CLAIM_VALUE);
+
+    return keycloakContainer;
+  }
+
+  private void createEntities(final Class<?> testClass, final Boolean shouldSetupKeycloak) {
     final var users = findUsers(testClass, null, ModifierSupport::isStatic);
+    final var mappings = findMappings(testClass, null, ModifierSupport::isStatic);
     final var groups = findGroups(testClass, null, ModifierSupport::isStatic);
     final var roles = findRoles(testClass, null, ModifierSupport::isStatic);
-    entityManager.withUser(users).withGroups(groups).withRoles(roles).await();
+    entityManager
+        .withUser(users)
+        .withMappings(mappings)
+        .withGroups(groups)
+        .withRoles(roles)
+        .await();
     users.forEach(
         user ->
             authenticatedClientFactory.createClientForUser(applicationUnderTest.application, user));
+
+    mappings.forEach(
+        mapping -> {
+          authenticatedClientFactory.createClientForMapping(
+              applicationUnderTest.application, mapping);
+
+          if (shouldSetupKeycloak) {
+            setupUserInKeycloak(mapping.id(), mapping.claimValue());
+          }
+        });
+  }
+
+  private void setupUserInKeycloak(final String mappingId, final String claimValue) {
+    final var clientRepresentation = new ClientRepresentation();
+    clientRepresentation.setClientId(claimValue);
```

Comment:
❓ Is it expected that we use the same value for this client ID and also the user id?

Reply:
@Ben-Sheppard I don't know, you tell me if that's realistic/expected 😄 I was struggling with the setup. Making it all the same value helped my to be authorized, and actually receive the claim value in the token.